src/main/java/org/kohsuke/github/GHRepository.java

```diff
@@ -125,8 +125,11 @@ public GHRepository() {
     private GHRepository source, parent;
 
     private Boolean isTemplate;
+
     private boolean compareUsePaginatedCommits;
 
+    private Set<GHRepositoryCustomProperty> custom_properties;
```

Comment:
Thoughts:  I'm not sure that `Set` helps you here.  You might want to represent this as
* `List` - getting all of them
* `Map<String, String>` - the docs seem to indicate that you can't have multiple properties with the same `name` value on a repository or organization.  And the properties are just name/value pairs...
* `Map<String,CustomProperty>` - This might be a good choice for future proofing, in case GitHub decides to make CustomProperty records more complex in future.

For either of the `Map` options you need to check whether `key` value uniqueness case-insensitive, case-sensitive but restricted to ASCII, or literary any Unicode/UTF-8 String.

Reply:
I got my custom properties returned to me using this code base referenced in a console app.

I was discussing map options with a friend of mine but haven't been able to get it to work yet. I am not familiar with how this repo handles object mapping and kept running into null errors.